mindmeld/auto_annotator.py

```diff
@@ -0,0 +1,708 @@
+
+        Returns:
+            entity (dict): A resolved entity dict or None if the entity isn't resolved.
+        """
+        entity["dim"] = self.ANNOTATOR_TO_DUCKLING_ENTITY_MAPPINGS[entity["dim"]]
+
+        candidates = self.duckling.get_candidates_for_text(entity["body"])
+        if len(candidates) == 0:
+            return
+
+        for candidate in candidates:
+            if (
+                candidate["entity_type"] == entity["dim"]
+                and entity["body"] == candidate["body"]
+            ):
+                entity["value"] = candidate["value"]
+                return entity
+
+    def _resolve_quantity(self, entity):
+        """ Resolves a quantity related entity. First looks for an exact match, then
+        for the largest substring match. Order of priority is "sys_distance" then "sys_quantity".
+        Unresolved entities are labelled as "sys_other-quantity"
+
+        Args:
+            entity (dict): A dictionary representing an entity.
+
+        Returns:
+            entity (dict): A resolved entity dict or None if the entity isn't resolved.
+        """
+        candidates = self.duckling.get_candidates_for_text(entity["body"])
+        if len(candidates) == 0:
+            entity["dim"] = "sys_other-quantity"
+            return entity
+
+        entity_types = ["distance", "quantity"]
+        for entity_type in entity_types:
+            for candidate in candidates:
+                if (
+                    candidate["dim"] == entity_type
+                    and candidate["body"] == entity["body"]
+                ):
+                    entity["value"] = candidate["value"]
+                    entity["dim"] = self.ANNOTATOR_TO_DUCKLING_ENTITY_MAPPINGS[
+                        entity_type
+                    ]
+                    return entity
+
+        if SpacyAnnotator._resolve_largest_substring(
+            entity, candidates, entity_types=entity_types, is_time_related=False
+        ):
+            return entity
+        else:
+            entity["dim"] = "sys_other-quantity"
+            return entity
+
+    def _resolve_percent(self, entity):
+        """ Resolves an entity related to percentage. Uses a heuristic of finding
+        the largest candidate value and dividing by 100.
+
+        Args:
+            entity (dict): A dictionary representing an entity.
+
+        Returns:
+            entity (dict): A resolved entity dict or None if the entity isn't resolved.
+        """
+        entity["dim"] = self.ANNOTATOR_TO_DUCKLING_ENTITY_MAPPINGS[entity["dim"]]
+
+        candidates = self.duckling.get_candidates_for_text(entity["body"])
+        if len(candidates) == 0:
+            return
+
+        possible_values = []
+        for candidate in candidates:
+            if candidate["entity_type"] == "sys_number":
+                possible_values.append(candidate["value"]["value"])
+        entity["value"]["value"] = max(possible_values) / 100
```

Comment:
Does this have to be a float or int?

Reply:
It would be a float. For example, if the query was "thirty five percent" and the highest found value was 35. It will take that value and divide by 100 to give 0.35 as the identified percent-type entity value

Reply:
What scenarios is this supposed to handle? Does it give the expected result for something like "35.85%"?

Reply:
@KKsharma99 - have you addressed this issue?

Reply:
Yes, the current code correctly resolves "35.85%" to 0.3585

```
        possible_values = []
        for candidate in candidates:
            if candidate["entity_type"] == "sys_number":
                value = candidate["value"]["value"]
                if isinstance(value, float):
                    entity["value"]["value"] = value / 100
                    return entity
                else:
                    possible_values.append(value)
        entity["value"]["value"] = max(possible_values) / 100
        return entity
```